src/app/directives/common/new-feature-indicator/new-feature-indicator.service.ts

```diff
@@ -0,0 +1,33 @@
+import { EventEmitter, Injectable } from '@angular/core';
+import { Store, select } from '@ngrx/store';
+import { NewFeatureIndicator } from 'app/directives/common/new-feature-indicator/new-feature-indicator.interface';
+import { AppState } from 'app/store';
+import { shownNewIndicatorKeysUpdated } from 'app/store/preferences/preferences.actions';
+import { selectPreferencesState } from 'app/store/preferences/preferences.selectors';
+
+@Injectable()
+export class NewFeatureIndicatorService {
+  onShown = new EventEmitter<NewFeatureIndicator>();
+  private shownIndicatorKeys: string[] = [];
+
+  constructor(
+    private store$: Store<AppState>,
+  ) {
+    this.store$.pipe(select(selectPreferencesState)).subscribe((prefs) => {
```

Comment:
Since this is not a singleton class it requires the until destroyed pipe and decorator.